core/controllers/skill_editor_test.py

```diff
@@ -96,47 +157,134 @@ def test_editable_skill_handler_put(self):
+            # skill.
+            update_skill_swap = self.swap(
+                skill_services, 'update_skill', self._mock_update_skill)
+            with update_skill_swap:
+                self.put_json(self.url, self.put_payload, csrf_token=csrf_token,
+                              expect_errors=True, expected_status_int=400)
+        self.logout()
 
-    def test_editable_skill_handler_delete(self):
+    def test_editable_skill_handler_delete_succeeds(self):
+        self.login(self.ADMIN_EMAIL)
         with self.swap(feconf, 'ENABLE_NEW_STRUCTURES', True):
             # Check that admins can delete a skill.
-            self.login(self.ADMIN_EMAIL)
-            self.delete_json(
-                '%s/%s' % (
-                    feconf.SKILL_EDITOR_DATA_URL_PREFIX, self.skill_id),
-                expected_status_int=200)
-            self.logout()
+            self.delete_json(self.url)
+        self.logout()
 
-            # Check that non-admins cannot delete a skill.
-            self.login(self.NEW_USER_EMAIL)
-            response = self.testapp.delete(
-                '%s/%s' % (
-                    feconf.SKILL_EDITOR_DATA_URL_PREFIX, self.skill_id),
-                expect_errors=True)
-            self.assertEqual(response.status_int, 401)
-            self.logout()
+    def test_editable_skill_handler_delete_fails(self):
+        self.login(self.ADMIN_EMAIL)
+        with self.swap(feconf, 'ENABLE_NEW_STRUCTURES', True):
+            # Check DELETE returns 404 when new strutures' pages are not
+            # enabled.
+            with self.swap(feconf, 'ENABLE_NEW_STRUCTURES', False):
+                self.delete_json(self.url, expect_errors=True,
+                                 expected_status_int=404)
+        # Check DELETE returns 500 when the skill still has associated
+        # questions.
+        feconf_swap = self.swap(feconf, 'ENABLE_NEW_STRUCTURES', True)
+        skill_has_questions_swap = self.swap(
+            skill_services, 'skill_has_associated_questions', lambda x: True)
+        with feconf_swap, skill_has_questions_swap:
+            self.delete_json(self.url, expect_errors=True,
+                             expected_status_int=500)
+        self.logout()
+
+
+class SkillPublishHandlerTest(BaseSkillEditorControllerTest):
+    """Tests for SkillPublishHandler."""
+
+    def setUp(self):
+        super(SkillPublishHandlerTest, self).setUp()
+        self.url = '%s/%s' % (feconf.SKILL_PUBLISH_URL_PREFIX, self.skill_id)
+
+    def test_skill_publish_handler_succeeds(self):
+        self.login(self.ADMIN_EMAIL)
+        with self.swap(feconf, 'ENABLE_NEW_STRUCTURES', True):
+            # Check that an admin can publish a skill.
+            csrf_token = self._get_csrf_token_for_put()
+            self.put_json(self.url, {'version': 1}, csrf_token=csrf_token)
+        self.logout()
+
+    def test_skill_publish_handler_fails(self):
+
+        def _publish_skill(*_args):
```

Comment:
See note above re naming for clarity. Also maybe extract this into a class-level method.

Reply:
Done.